spring-integration-redis/src/main/java/org/springframework/integration/redis/inbound/ReactiveRedisStreamMessageProducer.java

```diff
@@ -0,0 +1,207 @@
+import org.springframework.integration.redis.support.RedisHeaders;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.lang.Nullable;
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+import reactor.core.publisher.Flux;
+
+/**
+ * A {@link MessageProducerSupport} for Redis that reads message from a Redis Stream and publishes it to the provided
+ * output channel.
+ * By default this adapter reads message as a standalone client {@code XREAD} Redis command but we can switch to a
+ * Consumer Group feature {@code XREADGROUP} by setting {@link #consumerGroupName} and {@link #consumerName} fields.
+ *
+ * @author Attoumane Ahamadi
+ *
+ * @since 5.4
+ */
+public class ReactiveRedisStreamMessageProducer extends MessageProducerSupport {
+
+	private final ReactiveRedisConnectionFactory reactiveConnectionFactory;
+
+	private final Expression streamKeyExpression;
+
+	private EvaluationContext evaluationContext;
+
+	private StreamReceiver.StreamReceiverOptions<String, ?> streamReceiverOptions = StreamReceiver
+			.StreamReceiverOptions.builder().pollTimeout(Duration.ZERO).build();
+
+	private StreamReceiver<String, ?> streamReceiver;
+
+	private ReadOffset readOffset = ReadOffset.lastConsumed();
+
+	private boolean extractPayload = true;
+
+	private String streamKey;
+
+	@Nullable
+	private String consumerGroupName;
+
+	@Nullable
+	private String consumerName;
+
+	private boolean createConsumerGroupIfNotExist;
+
+	public ReactiveRedisStreamMessageProducer(ReactiveRedisConnectionFactory reactiveConnectionFactory,
+			String streamKey) {
+		this(reactiveConnectionFactory, new LiteralExpression(streamKey));
+	}
+
+	public ReactiveRedisStreamMessageProducer(ReactiveRedisConnectionFactory connectionFactory,
+			Expression streamKeyExpression) {
+		this.reactiveConnectionFactory = connectionFactory;
+		this.streamKeyExpression = streamKeyExpression;
+	}
+
+	/**
+	 * Define the offset from which we want to read message. By defaut the {@link ReadOffset#lastConsumed()} is used.
+	 * @param readOffset the desired offset
+	 */
+	public void setReadOffset(ReadOffset readOffset) {
+		this.readOffset = readOffset;
+	}
+
+	/**
+	 * Configure this channel adapter to extract the message payload.
+	 * @param extractPayload default true
+	 */
+	public void setExtractPayload(boolean extractPayload) {
+		this.extractPayload = extractPayload;
+	}
+
+	/**
+	 * Set the name of the Consumer Group. Is it possible to create that Consumer Group if desired, see:
+	 * {@link #createConsumerGroupIfNotExist}. Note that if this value is set, {@link #consumerName} should also be set
+	 * @param consumerGroupName the Consumer Group which this adapter should register to listen messages.
+	 */
+	public void setConsumerGroupName(@Nullable String consumerGroupName) {
+		this.consumerGroupName = consumerGroupName;
```

Comment:
We had a chat with Gary and decided to make the group name based on a `beanName`.
Therefore we only going to request a `consumerName` and only if that is provided we switch to a consumer group receiving. This convention therefore has to be documented on the class level JavaDocs and that `setConsumerName()`. And eventually in the docs, of course.

Reply:
So, after more thoughts Gary shared with me an idea that users may use the same application as different instances for different groups.

So, let's leave this setter, but still let's try to make a default one based on bean id!

Reply:
I'm involved in the development  of this feature because I really need it in my project. I planned to use it like this:
- `consumerGroupName = ${spring.application.name}`
- `consumerName = ${spring.application.name}-${instanceNumber}` (I am still wondering how to get the `instanceNumber` for an application deployed in Kubernetes? ^^)

@artembilan What do you mean by making a default setter based on bean id? Should I initialize `consumerGroupName` using the bean id? (I mean in the `onInit()` method, setting `consumerGroupName` if not provided by the client).

Reply:
That's true. In the `onInit()` you do `this.consumerGroupName = getBeanName()` if `consumerName` is provided and `this.consumerGroupName == null`.

I'm not familiar with Kubernetes and it fully not related to this project. 
I think what you do with Spring Boot properties is OK, only need to figure out what is there for that `instanceNumber` which I assume you borrowed from PCF.